Cython/Compiler/PyrexTypes.py

```diff
@@ -4929,9 +4929,10 @@ def independent_spanning_type(type1, type2):
             type1 = type1.ref_base_type
         else:
             type2 = type2.ref_base_type
-    if type1 == type2:
+    if type1.resolve() == type2.resolve():
```

Comment:
It's probably worth assigning these two to variables `resolved_type1` and `resolved_type2`.